scripts/generate-regression-tests.sh

```diff
@@ -72,136 +72,153 @@ which git > /dev/null || err "git not available"
         err "Unable to use $KORE/evm-semantics directory"
-    (cd $EVM_SEMANTICS && git submodule update --init --recursive)
 else
-    EVM_SEMANTICS=$(realpath ${EVM_SEMANTICS})
+    EVM_SEMANTICS=$(realpath "${EVM_SEMANTICS}")
     log "Using directory ${EVM_SEMANTICS} for evm-semantics" info
-    [ -f "$EVM_SEMANTICS/include/kframework/evm.md" ] || \
+    [ -f "$EVM_SEMANTICS/kevm-pyk/src/kevm_pyk/kproj/evm-semantics/evm.md" ] || \
         err "Provided evm-semantics directory '${EVM_SEMANTICS}' appears damaged"
 fi
 
 # determine K version to use from $KORE/deps/k, unless provided
 # deps/k_release contains a version number, the tag has a prefix 'v'
 if [ -z "${K_VERSION}" ]; then
-    [ -f $KORE/deps/k_release ] && K_VERSION=v$(cat $KORE/deps/k_release)
+    [ -f "$KORE"/deps/k_release ] && K_VERSION=v$(cat "$KORE"/deps/k_release)
     log "K version set to ${K_VERSION}"
 fi
 
 # build evm-semantics
 log "Checking out and building ${EVM_VERSION:-latest master} in ${EVM_SEMANTICS}"
-cd ${EVM_SEMANTICS}
+cd "${EVM_SEMANTICS}"
 git fetch
-git checkout ${EVM_VERSION:-master}
-git pull
+git checkout "${EVM_VERSION:-master}"
 git submodule update --init --recursive
-log "Manually setting K dependency to ${K_VERSION}"
 
-log "Cleaning ${EVM_SEMANTICS}"
-make clean
+log "Checking out and building K ${K_VERSION} in ${EVM_SEMANTICS}/deps/k"
 # if we are under a nix develop shell (NIX variable set), don't build K
 if [ -z "$NIX" ]; then
-    (cd deps/k && git checkout ${K_VERSION} && git submodule update --init --recursive)
-    make deps
-    export PATH=$(pwd)/.build/usr/lib/kevm/kframework/bin:$PATH
+    cd deps
+    if [ ! -d k ]; then
+        git clone git@github.com:runtimeverification/k.git
+    fi
+    cd k
+    git fetch
+    git checkout "$K_VERSION"
+    git submodule update --init --recursive
+    mvn package -DskipTests || err "Unable to build K ${K_VERSION}"
+    PATH=$(pwd)/k-distribution/target/release/k/bin:$PATH
+    export PATH
     which kore-exec
+    cd "${EVM_SEMANTICS}"
 else
     log "Testing nix-provided K (kompile --version)"
-    kompile --version && log "(^^^ nix-provided K ^^^)" || err "K unavailable but NIX=$NIX"
+    kompile --version || err "K unavailable but NIX=$NIX"
 fi
+
 log "Building evm-semantics with dependencies"
-make plugin-deps poetry kevm-pyk
-export PATH=$(pwd)/.build/usr/bin:$PATH
-make build-kevm build-haskell
+make poetry
+#poetry -C kevm-pyk run kevm-dist clean
+poetry -C kevm-pyk run kevm-dist build plugin
+poetry -C kevm-pyk run kevm-dist build -j"$(nproc)"
 
 kollect() {
     local name="$1"
 
-    local archive=kevm-bug-$name.tar.gz
     local script=test-$name.sh
     local def=test-$name-definition.kore
     local spec=test-$name-spec.kore
     local tmp=$name-tmp
 
-    cd ${EVM_SEMANTICS}
+    pushd "${EVM_SEMANTICS}"/"$name"
 
-    mkdir $tmp
-    mv $archive $tmp
-    cd $tmp
-    tar -xf $archive
+    mkdir "$tmp"
+    cp kevm-bug-*.tar.gz "$tmp"
```

Comment:
The `kevm-bug-*.tar.gz` glob here is to work around some weird behavior where `functional/lemmas` generates a bug report named `kevm-bug-lemma.tar.gz` (without the "s"). I updated the script to run each test in its own directory, so the glob should find a unique file.